From 9ca08d8781150a6157c1f8824f2f6b7f36997ecc Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sat, 12 Sep 2020 12:32:50 +0200
Subject: [PATCH 13/26] No longer make re-assigning legacy audio groups fatal.
 Mi9 declares AUDIO_STREAM_PATCH and AUDIO_STREAM_REROUTING which is defined
 by framework too

Change-Id: I794fe22d63a8af705be4f5f09b9879ecaab3eae5
---
 services/audiopolicy/engine/common/src/EngineBase.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/services/audiopolicy/engine/common/src/EngineBase.cpp b/services/audiopolicy/engine/common/src/EngineBase.cpp
index 99507eee5d..e31cc21b17 100644
--- a/services/audiopolicy/engine/common/src/EngineBase.cpp
+++ b/services/audiopolicy/engine/common/src/EngineBase.cpp
@@ -222,9 +222,9 @@ engineConfig::ParsingResult EngineBase::loadAudioPolicyEngineConfig()
             }
             if (group.stream != AUDIO_STREAM_DEFAULT) {
                 // A legacy stream can be assigned once to a volume group
-                LOG_ALWAYS_FATAL_IF(checkStreamForGroups(group.stream, mVolumeGroups),
-                                    "stream %s already assigned to a volume group, "
-                                    "review the configuration", toString(group.stream).c_str());
+                if(checkStreamForGroups(group.stream, mVolumeGroups)) {
+                    ALOGE("stream %s already assigned to a volume group, review the configuration", toString(group.stream).c_str());
+                }
                 volumeGroup->addSupportedStream(group.stream);
             }
             addSupportedAttributesToGroup(group, volumeGroup, strategy);
-- 
2.34.1

